Port group API support

anfredette opened the pull request:
ovn-kubernetes needs the ability to add ACLs to port groups, but the current ACL APIs only support logical switches, and there are no port group APIs. This PR adds port group API support and tests.

anfredette commented:
@hzhou8 @abhat @vjayaramrh please review.

vtolstov reviewed (approved):
LGTM

hzhou8 commented on client.go:
```diff
@@ -206,6 +206,15 @@ type Client interface {
+	// Deletes port group "group". It is an error if "group" does not exist.
+	PortGroupDel(group string) (*OvnCommand, error)
+	// Get PortGroup data structure if it exists
+	PortGroupGet(group string) ([]*PortGroup, error)
```
Port_Group's name column is unique (index), so I think it doesn't make sense to return an array. It is better to return a pointer - nil if doesn't exist.

anfredette replied:
Will do.

hzhou8 commented on client.go:
```diff
@@ -206,6 +206,15 @@ type Client interface {
+	// Creates a new port group in the Port_Group table named "group" with optional "ports" added to the group.
+	PortGroupAdd(group string, ports []string, external_ids map[string]string) (*OvnCommand, error)
+	// Sets "ports" and/or "external_ids" on the port group named "group". It is an error if group does not exist.
+	PortGroupSet(group string, ports []string, external_ids map[string]string) (*OvnCommand, error)
```
For most APIs of this library, the naming convention is "Update" for the whole row, "Set" for a specific column. Since this API updates multiple columns, it may be better named as PortGroupUpdate(...).

In addition, there is a column ACL. Would you want to include in this API so that it can be updated? Same for PortGroupAdd() API.

anfredette replied:
I'll update the name. I was intending to just add the API support for the existing port-group functionality in this PR, and then add support for the ACL column when I added the generic ACL support that we discussed so I could test them together.  Is this okay with you?

hzhou8 commented on port_group_test.go:
```diff
@@ -17,256 +17,367 @@
-	PG_TEST_OPT_1_VAL_2    = "baz1"
-	PG_TEST_OPT_2_KEY      = "foo2"
-	PG_TEST_OPT_2_VAL      = "bar2"
+	PG_TEST_PG1         = "TestPortGroup1"
```
This commit seems to be a fix/revision of the previous one. I think it is better to be combined.

anfredette replied:
Yup. I'll squash them.

hzhou8 reviewed (changes requested):
Thanks @anfredette for helping on this. Please see my comments.

anfredette commented:
Vasily and Han, thanks for the review!

anfredette commented:
I think I've addressed all of the comments, except for adding support for the ACL column which I plan to take care of when I add ACL support for port-groups.

vtolstov commented on port_group.go:
```diff
@@ -43,11 +43,19 @@ func (odbi *ovndb) pgAddImp(group string, ports []string, external_ids map[strin
-		return nil, err
+
+	if ports != nil {
+		var portUUIDs []libovsdb.UUID
```
portUUIDs := make([]libovsdb.UUID, 0, len(ports))

vtolstov commented on port_group.go:
```diff
@@ -66,18 +74,30 @@ func (odbi *ovndb) pgAddImp(group string, ports []string, external_ids map[strin
-	row["ports"] = pgports
+
+	if ports != nil {
+		var portUUIDs []libovsdb.UUID
```
portUUIDs := make([]libovsdb.UUID, 0, len(ports))

vtolstov reviewed (commented):
minor comments about optimizations, lgtm

anfredette commented:
@vtolstov, thanks for the suggestion.  I'll make that optimization in my next PR.
